plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/portforwading/PortForwardingWizardPage.java

```diff
@@ -0,0 +1,273 @@
+	}
+
+	@Override
+	protected void doCreateControls(Composite parent, DataBindingContext dbc) {
+		GridLayoutFactory.fillDefaults().margins(6, 6).applyTo(parent);
+		Composite container = new Composite(parent, SWT.NONE);
+		GridDataFactory.fillDefaults().align(SWT.FILL, SWT.FILL).grab(true, true).applyTo(container);
+		GridLayoutFactory.fillDefaults().numColumns(2).applyTo(container);
+
+		Composite tableContainer = new Composite(container, SWT.NONE);
+		this.viewer = createTable(tableContainer, dbc);
+		GridDataFactory.fillDefaults().span(1, 3).align(SWT.FILL, SWT.FILL).grab(true, true).applyTo(tableContainer);
+
+		startButton = new Button(container, SWT.PUSH);
+		startButton.setText("Start All");
+		GridDataFactory.fillDefaults().hint(110, SWT.DEFAULT).align(SWT.FILL, SWT.TOP).applyTo(startButton);
+		startButton.addSelectionListener(onStartPortForwarding());
+
+		stopButton = new Button(container, SWT.PUSH);
+		stopButton.setText("Stop All");
+		GridDataFactory.fillDefaults().hint(110, SWT.DEFAULT).align(SWT.FILL, SWT.TOP).applyTo(stopButton);
+		stopButton.addSelectionListener(onStopPortForwarding());
+
+		final Button findFreesPortButton = new Button(container, SWT.CHECK);
+		findFreesPortButton.setText("Find free local ports for remote ports");
+		GridDataFactory.fillDefaults().span(2, 1).align(SWT.FILL, SWT.CENTER).grab(false, false)
+				.applyTo(findFreesPortButton);
+		final IObservableValue findFreePortsButtonObservable = BeanProperties.value(
+				PortForwardingWizardModel.PROPERTY_USE_FREE_PORTS).observe(wizardModel);
+		final IObservableValue findFreePortsButtonSelection = WidgetProperties.selection().observe(findFreesPortButton);
+		dbc.bindValue(findFreePortsButtonSelection, findFreePortsButtonObservable);
+		findFreePortsButtonObservable.addValueChangeListener(new IValueChangeListener() {
+			@Override
+			public void handleValueChange(ValueChangeEvent event) {
+				refreshViewerInput(wizardModel.getForwardablePorts());
+			}
+		});
+
+		// enabling/disabling controls
+		IObservableValue portForwardingStartedObservable = BeanProperties.value(
+				PortForwardingWizardModel.PROPERTY_PORT_FORWARDING).observe(wizardModel);
+		
+		
+		ValueBindingBuilder.bind(WidgetProperties.enabled().observe(startButton))
+			.notUpdating(portForwardingStartedObservable).converting(new InvertingBooleanConverter()).in(dbc);
+
+		ValueBindingBuilder.bind(WidgetProperties.enabled().observe(stopButton))
+				.notUpdating(portForwardingStartedObservable).in(dbc);
+		
+		ValueBindingBuilder.bind(WidgetProperties.enabled().observe(findFreesPortButton))
+				.notUpdating(portForwardingStartedObservable).converting(new InvertingBooleanConverter()).in(dbc);
+
+	}
+
+
+	private SelectionListener onStartPortForwarding() {
+		return new SelectionAdapter() {
+			@Override
+			public void widgetSelected(SelectionEvent e) {
+				try {
+					WizardUtils.runInWizard(new Job("Starting port forwarding...") {
```

Comment:
ah, here's the job. My comments further up might be bogus then. Right?
So if you tell the wizard to require a progress bar (wizard#setNeedsProgressMonitor(true) we should get a progress bar while this job is running.